### PR TITLE
LVNODCDXI-72

### PR DESCRIPTION
**Pull request by mr-redFox**
added refresh token in middleware

### CODE REVIEW

**Review by GeFeSTTS — DISMISSED** [review]

**Review by skiffua — DISMISSED** [review]

**Review by Kindest13 — DISMISSED** [review]

File: src/server/helpers/token.js [hunk 1]
```diff
@@ -8,6 +8,10 @@ dotenv.config();
 const { SECRET } = process.env;
 const tokenLife = process.env.TOKEN_LIFE || '30m';
 
+export const verify = async token => jwt.verify(token, SECRET);
```
Comment (tonypntsrn) [inline src/server/helpers/token.js]: Do you need this functions to be async?

File: src/server/middlewares/auth.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      await Redis.putToken(newTokens.refresh, userId);
+    }
+
+    return newTokens;
+  } catch (error) {
+    throw error;
+  }
+};
+
+export default async (req, res, next) => {
```
Comment (tonypntsrn) [inline src/server/middlewares/auth.js]: Do you export anonymous function?

File: src/server/helpers/token.js [hunk 3]
```diff
@@ -8,6 +8,10 @@ dotenv.config();
...
+
+export const decode = async token => jwt.decode(token);
```
Comment (tonypntsrn) [inline src/server/helpers/token.js]: I would suggest using better name for functions

File: src/server/helpers/token.js (hunk 1 above)
Comment (mr-redFox) [reply src/server/helpers/token.js]: done

File: src/server/middlewares/auth.js (hunk 2 above)
Comment (mr-redFox) [reply src/server/middlewares/auth.js]: added name for this function.
Done

File: src/server/helpers/token.js (hunk 3 above)
Comment (mr-redFox) [reply src/server/helpers/token.js]: it just method overloading base jwt lib. I used this methods like Token.verify()

**Review by GeFeSTTS — APPROVED** [review]

**Review by tonypntsrn — APPROVED** [review]

**Pull request merged**